### DISCUSSION TITLE
Trying to set up Nextjs, Tailwind2 and Storybook with Yarn Berry

### DISCUSSION THREAD

**Discussion by jonparkdev** [storybookjs/storybook#15961] [Help]
I am trying to set up a nextjs app with tailwind and storybook storybook. I have scoured the issues thread for a solution and found some familiar issues but none of the solutions have yet to solve my issue. I will reproduce my set steps below. Thanks!

## 1. Set up Yarn Berry and Basic Monorepo setup

```$ yarn set version berry```
```$ yarn init```

#### In `package.json` set up workspaces
```json
{
  "name": "my-project",
  "packageManager": "yarn@3.0.1",
  "workspaces": [
    "packages/**"
  ],
  "private": true
}
```

```$ mkdir packages```

**Also run these commands to install Typescript at the base of the monorepo (for reason's outside the scope of this)**

```$ yarn add -D typescript@~4.3```
```$ yarn dlx @yarnpkg/sdks vscode``` (if you use vscode, @see https://yarnpkg.com/getting-started/editor-sdks for other IDE's)

## 2. set up nextjs

```$ cd packages ```
```$ yarn create next-app --typescript```

**name the app whatever, I went with the default, `my-app`**

## 3. set up storybook 

```$ yarn dlx -p @storybook/cli sb init```

**By this point if you run yarn storybook all should be still working** 

## 4. Set up tailwind

```$ yarn add -D tailwindcss@latest postcss@latest autoprefixer@latest```
```$ npx tailwindcss init -p```

**Nextjs uses Postcss8 and Storybook Postcss7, so we need to make some changes to Storybook**

```$ yarn add -D @storybook/addon-postcss```
Update the /packages/my-app/.storybook/main. Should look like:

```
module.exports = {
  stories: [
    "../stories/**/*.stories.mdx",
    "../stories/**/*.stories.@(js|jsx|ts|tsx)",
  ],
  addons: [
    "@storybook/addon-links",
    "@storybook/addon-essentials",
    {
      name: "@storybook/addon-postcss",
      options: {
        postcssLoaderOptions: {
          implementation: require("postcss"),
        },
      },
    },
  ],
};
```

Now if you try to run yarn storybook, you get a module error message. the tldr is
```Loading PostCSS "tailwindcss" plugin failed: postcss-loader tried to access tailwindcss, but it isn't declared in its dependencies; this makes the require call ambiguous and unsound.```

So I suspect that the storybook is using the `postcss.config.js` generated by tailwind

#### 1. adding a webpackFinal to the /packages/my-app/.storybook/main to seperatly incorperate the postcss-loader, so: 
```
const path = require("path");

module.exports = {
  stories: [
    "../stories/**/*.stories.mdx",
    "../stories/**/*.stories.@(js|jsx|ts|tsx)",
  ],
  addons: [
    "@storybook/addon-links",
    "@storybook/addon-essentials",
    {
      name: "@storybook/addon-postcss",
      options: {
        postcssLoaderOptions: {
          implementation: require("postcss"),
        },
      },
    },
  ],
  webpackFinal: async (config) => {
    config.module.rules.push({
      test: /\.css$/,
      use: [
        {
          loader: "postcss-loader",
          options: {
            postcssOptions: {
              plugins: [require("tailwindcss"), require("autoprefixer")],
            },
          },
        },
      ],
      include: path.resolve(__dirname, "../"),
    });

    return config;
  },
};
```

#### 2. Some threads talk about using postcss7 compatible tailwind packages and removing the @storybook/postcss-addon, so
```npm install -D tailwindcss@npm:@tailwindcss/postcss7-compat postcss@^7 autoprefixer@^9```

#### 3. Downgrading @storybook/** packages to 6.2.4 (or other arbitrary v6.2.*)

Anyways, I felt I have spent quite a bit of time on this, so if anyone can provide me with some insight to what may be going on, it would greatly appreciated! 

Interestingly if you modify postcss.config.js from: 

```
module.exports = {
  plugins: {
    tailwindcss: {},
    autoprefixer: {},
  },
}
```
to ----->

```
module.exports = {
  plugins: [require("tailwindcss"), require("autoprefixer")]
}
```

Storybook will run but...

Nextjs will then be broken because

> If your postcss.config.js needs to support other non-Next.js tools in the same project, you must use the interoperable object-based format instead:

**Comment by jonparkdev** [comment]
For ease of set, here is demo repo:

https://github.com/jonparkdev/storybook-tailwind-nextjs-yarn2

**Comment by 7iomka** [comment]
@jonparkdev Same issue.. Anyone help?

**Comment by pencilcheck** [comment]
doesn't work still in 2022, but instead of nuxtjs I used nuxt. Tailwind still doesn't work. Reverted to yarn1 and it worked.
